### DISCUSSION TITLE
3.8.2

### DISCUSSION THREAD

**Discussion by Davidy22** [Guake/guake#2027] [Announcements]
## What's Changed
* Fix system font only being applied to last tab by @Davidy22 in https://github.com/Guake/guake/pull/1954
* Use keycode for keybindings by @vantu5z in https://github.com/Guake/guake/pull/1953
* Replace ｜ with non unicode dependent | by @Davidy22 in https://github.com/Guake/guake/pull/1966
* Update news file by @Davidy22 in https://github.com/Guake/guake/pull/1951
* Add pbr fallback by @Davidy22 in https://github.com/Guake/guake/pull/1964
* Fix vte spawn_sync runtime check failed by @mlouielu in https://github.com/Guake/guake/pull/1974
* Add `GUAKE_ENABLE_WAYLAND` environment variable by @VitalyAnkh in https://github.com/Guake/guake/pull/1975
* Add instructions for PPA install by @Davidy22 in https://github.com/Guake/guake/pull/1961
* Switch default for hide tab bar if only one tab to disabled by @Davidy22 in https://github.com/Guake/guake/pull/1983
* Changed default keybindig to toggle hide on lose focus. Fix #1719 by @jcfstabile in https://github.com/Guake/guake/pull/1988
* Fix module gi not found when make run  by @jcfstabile in https://github.com/Guake/guake/pull/1989
* Revert "Fix module gi not found when make run " by @mlouielu in https://github.com/Guake/guake/pull/1992
* Drop pbr by @eli-schwartz in https://github.com/Guake/guake/pull/1991
* Bump pipenv from 2021.5.29 to 2022.1.8 by @dependabot in https://github.com/Guake/guake/pull/2005
* Fix upload to pypi test by @Davidy22 in https://github.com/Guake/guake/pull/2002
* Fix translation file inclusion by @Davidy22 in https://github.com/Guake/guake/pull/2004
* feature: --is-visible option #1926 by @rjeb in https://github.com/Guake/guake/pull/2008
* Reinstate double click to open tabs by @Davidy22 in https://github.com/Guake/guake/pull/2010
* Fix deepsource complaints by @Davidy22 in https://github.com/Guake/guake/pull/2009
* ci: enable upload to pypi by @Davidy22 in https://github.com/Guake/guake/pull/2007
* Generate release notes for 3.8.2 by @Davidy22 in https://github.com/Guake/guake/pull/2012
* Update translation from Weblate by @gsemet in https://github.com/Guake/guake/pull/2013
* Revert "doc: update NEWS.rst" by @Davidy22 in https://github.com/Guake/guake/pull/2014
* ci: no pipeline on pull_request by @gsemet in https://github.com/Guake/guake/pull/2016
* Revert "ci: no pipeline on pull_request" by @Davidy22 in https://github.com/Guake/guake/pull/2018
* Fix arbitrary execution via dbus security flaw by @Davidy22 in https://github.com/Guake/guake/pull/2017
* Eliminate redundant terminal spawning by @Davidy22 in https://github.com/Guake/guake/pull/2020
* Generate release notes for 3.8.2 by @Davidy22 in https://github.com/Guake/guake/pull/2024

## New Contributors
* @VitalyAnkh made their first contribution in https://github.com/Guake/guake/pull/1975
* @jcfstabile made their first contribution in https://github.com/Guake/guake/pull/1988
* @rjeb made their first contribution in https://github.com/Guake/guake/pull/2008

**Full Changelog**: https://github.com/Guake/guake/compare/3.8.1...3.8.2

<hr /><em>This discussion was created from the release <a href='https://github.com/Guake/guake/releases/tag/3.8.2'>3.8.2</a>.</em>

**Comment by Darkle** [comment]
Will this release make it to pypi?
Thanks.

**Reply by gsemet** [reply]
Sûre within the day !

**Comment by Darkle** [comment]
It still seems to be 3.7.0 on pypi: https://pypi.org/project/guake/

**Comment by gsemet** [comment]
I have released 3.8.3 which basically replaces 3.8.2. I also have disabled merge commit because it seems to mess with how PBR computes version.

**Reply by Davidy22** [reply]
Still seems to be uploading weirdly, for now, since we clearly have valid wheels we can just manually upload the latest manually

**Reply by gsemet** [reply]
indeed, not fixed, because of setuptools_scm: https://github.com/Guake/guake/issues/2037

**Reply by eli-schwartz** [reply]
The release process seems to be modifying the source code locally, causing `git describe` to report a "dirty" version.

setuptools_scm then computes the version with the dev node and timestamp, because it isn't a clean release. But in my PR to implement setuptools_scm I configured the timestamp to not be used, because it didn't match the previous style.

What this means: the release process is trying to release `{tag}.dev0.dirty` with a slightly confusing name of `{tag}.dev0`, which in retrospect may not have been the most optimal configuration setting to use.

The best solution here is to fix the release workflow to not report unstaged changes. Alternatively I could take a look at fine-tuning the setuptools_scm config so that it doesn't report on dirty changes.

**Reply by eli-schwartz** [reply]
... and we cross-posted. :)

**Reply by gsemet** [reply]
i like actually the way setuptools_scm works, but with guake we still have dirty files even during normal build. Ideally we would take the github tag information as source of truth (if GITHUB_REF_TYPE==tag, then take GITHUB_REF_NAME for the version string)
